mantidproject/mantid
# Fix crash when closing plot window and focusing on another

This crash was introduced [here](https://github.com/mantidproject/mantid/blob/6beaf280849706bed2914d32e1b7372ff491d418/qt/applications/workbench/workbench/widgets/plotselector/view.py#L283) where a `raise RuntimeError` statement was added, but the [parent code](https://github.com/mantidproject/mantid/blob/6beaf280849706bed2914d32e1b7372ff491d418/qt/applications/workbench/workbench/widgets/plotselector/presenter.py#L143) was set up to except a `TypeError` (when `PlotSelectorView._get_row_and_widget_from_plot_number` would have returned `None` instead of a tuple before the `raise RuntimeError` statement was added).

This PR changes the error raised to a `ValueError`, and modifies the error handling code to deal with this where necessary.

**Report to:** Saurabh

**To test:**
- Check that the behaviour described on the original issue #31031 is fixed
- Check that opening, closing, re-odering etc. multiple plot windows does not throw errors.

Fixes #31031.




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
